xonsh/tools.py

```diff
@@ -1767,11 +1767,33 @@ def to_completion_mode(x):
     return y
 
 
-def is_str_str_dict(x):
-    """Tests if something is a str:str dictionary"""
-    return isinstance(x, dict) and all(
-        isinstance(k, str) and isinstance(v, str) for k, v in x.items()
-    )
+def is_tok_color_dict(x):
+    from pygments.token import _TokenType, string_to_tokentype
+
+    from xonsh.ptk_shell.shell import _style_from_pygments_dict
+
+    """Tests if something is a Token:Color dictionary"""
+    if not isinstance(x, dict):
+        return False
+    """Check if is a Token:str dict"""
+    for k, v in x.items():
+        if not isinstance(v, str):
+            return False
+        try:
+            k = _TokenType(k)
+            string_to_tokentype(k)
+        except (TypeError, AttributeError):
+            msg = f'"{k}" is not a valid Token.'
+            warnings.warn(msg, RuntimeWarning)
+            return False
+    """Check str is a valid color"""
+    try:
+        _style_from_pygments_dict(x)
+    except (AssertionError, AttributeError, TypeError, ValueError):
```

Comment:
Are all of these possible errors that get bubbled up from `_style_from_pygments_dict`?

Reply:
Yes, I found that the pygments library throws an AssertionError whenever it tries to convert an invalid string into a style. All other uses _style_from_pygments_dict in the project also catch a ValueError which I couldn’t find an input that would throw this error, but I assume its thrown with certain inputs that cant be converting to a style. The other two errors AttributeError and TypeError occur for invalid tokens buts after looking at the code, I think they can be removed since all tokens in the dictionary should have already been checked.